Wrong codegen with openArray

Two examples of the same behaviour:

## Example
```nim
import threadpool
proc take[T](f:openArray[T]) =
  echo f.len
let f = @[0,1,2,3,4]
spawn take(f.toOpenArray(0,4))
sync()
```

## Current Output
with ``nim c --gc:arc``:
```
5
```
with ``nim c``:
```
/home/bwsq/.cache/nim/test1_d/@mtest1.nim.c: In function ‘takeWrapper__dH6EykH7QyFx6AjSxIAxUQ’:
/home/bwsq/.cache/nim/test1_d/@mtest1.nim.c:201:75: error: expected ‘)’ before ‘Len_0’
  201 |  genericDeepCopyOpenArray((void*)(&colontmp_), (void*)(&T1_), (&colontmp_)Len_0, (&NTI__0HB6wOMHiJYheBmdOv6sWw_));
```

this is all on devel, on 1.2.6 the spawn example instead gives:
```
/home/bwsq/playground/gene/test1.nim(6, 13) Error: invalid context for 'toOpenArray'; 'toOpenArray' is only valid within a call expression
```



The following examples are not real issues perhaps, but are more ways that can produce the same behaviour.

 on 1.2.6:
#### Example
```nim
type
  Foo = object
    data: openArray[int]
proc take(f: Foo){.exportc.} =
  echo f.data.len
```
#### Output
```
/home/bwsq/.cache/nim/test1_d/@mtest1.nim.c: In function ‘take’:
/home/bwsq/.cache/nim/test1_d/@mtest1.nim.c:103:24: error: ‘Foo’ has no member named ‘dataLen_0’
  103 |  T1_[0] = nimIntToStr(f.dataLen_0);
```
On devel:
#### Example
```nim
{.experimental:"views".}
type
  Foo = object
    data: openArray[int]
let f = Foo(data: [1,2,3,4,5])
#echo f.data.len # 2 !!!
#echo f.data #sigsegv
echo f
```
#### Output
```
/home/bwsq/.cache/nim/test1_d/stdlib_dollars.nim.c: In function ‘dollar___hIlg9bjpg83oxdsanCK3aeA’:
/home/bwsq/.cache/nim/test1_d/stdlib_dollars.nim.c:285:58: error: ‘tyObject_Foo__BxyWqiopCSV4Y9aVz0qv0QQ’ has no member named ‘dataLen_0’
  285 |  addQuoted__5SWWWhU9aDJTbLI38NrK19aQ((&result), x.data, x.dataLen_0);
```
Wrapping the access in a proc 
```nim
proc take1(f:Foo) = echo f
proc take2(f:Foo) = echo f.data
proc take3(f:Foo) = echo f.data.len
```
Always results in 'no member named 'dataLen_0' errors ( no '2' or sigsegv )


```
$ nim -v
Nim Compiler Version 1.3.7 [Linux: amd64]
Compiled at 2020-09-29
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 50b6f6996ab604931d8ee7ca2a26bfdce5559933
active boot switches: -d:release
```

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -450,9 +450,10 @@ proc genDeepCopy(p: BProc; dest, src: TLoc) =
               [addrLoc(p.config, dest), rdLoc(src),
               genTypeInfoV1(p.module, dest.t, dest.lode.info)])
   of tyOpenArray, tyVarargs:
+    let source = addrLocOrTemp(src)
     linefmt(p, cpsStmts,
-         "#genericDeepCopyOpenArray((void*)$1, (void*)$2, $1Len_0, $3);$n",
-         [addrLoc(p.config, dest), addrLocOrTemp(src),
+         "#genericDeepCopyOpenArray((void*)$1, (void*)$2, $2->Field1, $3);$n",
+         [addrLoc(p.config, dest), source,
          genTypeInfoV1(p.module, dest.t, dest.lode.info)])
   of tySet:
     if mapSetType(p.config, ty) == ctArray:

--- a/tests/ccgbugs/t15428.nim
+++ b/tests/ccgbugs/t15428.nim
@@ -1,0 +1,22 @@
+discard """
+    cmd: "nim $target --mm:refc $file"
+    output: '''5
+5
+[1, 2, 3, 4, 5]
+(data: [1, 2, 3, 4, 5])
+'''
+"""
+
+proc take[T](f: openArray[T]) =
+  echo f.len
+let f = @[0,1,2,3,4]
+take(f.toOpenArray(0,4))
+
+{.experimental: "views".}
+type
+  Foo = object
+    data: openArray[int]
+let f2 = Foo(data: [1,2,3,4,5])
+echo f2.data.len
+echo f2.data
+echo f2

